### PR TITLE
Update libc::stat field names

### PR DESCRIPTION
**Pull request by ian-h-chamberlain**
Required due to https://github.com/Meziu/libc/pull/9#discussion_r805436215

@Meziu @AzureMarker

### CODE REVIEW

File: library/std/src/sys/unix/fs.rs [hunk 1]
```diff
@@ -321,14 +321,19 @@ impl FileAttr {
         }))
```
Comment (AzureMarker) [inline library/std/src/sys/unix/fs.rs:321]: The line above here (just out of reach of GitHub's review system) seems to suggest that maybe instead of holding a timespec we could hold a seconds and nanoseconds field and then easily re-use the common implementation here.

Though if the libc test says we should use a timespec, that's fine too.

Comment (ian-h-chamberlain) [reply library/std/src/sys/unix/fs.rs:321]: I forget the exact result from the test, let me try again and see. Consistency with other targets would probably be nicer if possible

Comment (ian-h-chamberlain) [reply library/std/src/sys/unix/fs.rs:321]: Ok, I found that some other targets effectively do some "renaming" of the the fields: https://github.com/Meziu/libc/blob/master/libc-test/build.rs#L3786

Based on that, I'm not sure which is more desirable, given that is a `FIXME` I'm not sure if they want to fix the test to include the unpacked fields, or if they want to rename the fields (would be a breaking change, presumably?)

Also, since this PR merged already, maybe just let it ride and see how libc maintainers feel when we upstream it?

Comment (AzureMarker) [reply library/std/src/sys/unix/fs.rs:321]: It looks like only that one "Haiku" target does that renaming, so I'm fine with leaving it as is and seeing what the libc maintainers think.

Comment (Meziu) [reply library/std/src/sys/unix/fs.rs:321]: I believe it's fine. They'll tell us if anything, but I don't see why it should be different for a new target.
